dev/NavigationView/readme.md

```diff
@@ -0,0 +1,51 @@
+# NavigationView Control
+
+NavigationView is a powerful control that makes it easy to integrate a rich navigation experience into apps.
+This document outlines overall structure and behaviors.
+Information related to the NavigationViewItem, or the rendering of the control see the documents in the /NavigationView/docs folder.
+
+## Architecture
+
+NavigationView is a complex control, which has a lot of internal states to store.
+
+Rendering of the menu items is being done through three ItemsRepeaters, one for the NavigationView in left mode, one for the NavigationView in top mode and one for the overflow menu.
+For more on this, see [the rendering document](./docs/rendering.md).
+
+The selection is being tracked through a SelectionModel, which uses IndexPath tracking for hierarchical NavigationViewItems.
```

Comment:
Maybe there's no need to specifically mention hierarchy here? As IndexPath is being used to track selection for all cases.

Reply:
Yeah, hierarchy seemed a bit redundant here, so I have removed it.